insecure/net/message_processor_test.go

```diff
@@ -0,0 +1,58 @@
+package net
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	mockinsecure "github.com/onflow/flow-go/insecure/mock"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/libp2p/message"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/onflow/flow-go/utils/unittest/network"
+)
+
+// TestProcess_AttackerRegistered checks that when an attacker is registered with a corrupt network, the corrupt
+// message processor does NOT send the message to the original message processor.
+// This test simulates the attacker not being registered by making HandleIncomingEvent() return true.
+func TestProcess_AttackerRegistered(t *testing.T) {
```

Comment:
Updated in https://github.com/onflow/flow-go/pull/3255/commits/e1112cacbf1c96c93ef859401f365a56555135d4

I didn't use `originalProcessor.AssertNotCalled(t, "Process", channel, originId, msg)` because the test would fail without that line if the original message processor was called so that assertion was never used.